src/intrprtr.h

```diff
@@ -22,40 +22,34 @@
 #include "common.h"
 #include "gap.h"
 
-/****************************************************************************
-**
-*V  IntrIgnoring  . . . . . . . . . interpreter is currently ignoring actions
-**
-**  If 'IntrIgnoring'  is  non-zero,  the interpreter  is  currently ignoring
-**  actions.  The interpreter switches to this mode for  the right operand of
-**  'or' and 'and'  constructs where the  left operand already determines the
-**  outcome.
-**
-**  This mode is also used in Info and Assert, when arguments are not printed. 
-*/
-/* TL: extern UInt IntrIgnoring; */
 
+struct IntrState {
+
+    // If 'IntrIgnoring' is non-zero, the interpreter is currently ignoring
+    // actions. The interpreter switches to this mode for the right operand of
+    // 'or' and 'and' constructs where the left operand already determines the
+    // outcome.
+    //
+    // This mode is also used in Info and Assert, when arguments are not
+    // printed.
+    UInt ignoring;
+
+    // If 'coding' is non-zero, the interpreter is currently coding actions.
+    // The interpreter switches to this mode for constructs that it cannot
+    // directly interpret, such as loops or function bodies.
+    UInt coding;
+
+    // If 'IntrReturning' is non-zero, the interpreter is currently exiting
+    // statements enclosing a return statement. Actions from these statements
+    // are ignored.
+    ExecStatus returning;
+
+    // 'StackObj' is the stack of values.
+    Obj StackObj;
```

Comment:
The above four variables were moved from `GAPState` into this new `struct IntrState`. I reformatted the comments suitable, and stripped irrelevant bits from them.

Reply:
There is no such type in GAP yet, that PR is still nit merged